NXDOC-2788: Add Search Service Upgrade page (#2419)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-search-service.md

```

## Page Providers

Any reference to `ElasticSearchNxqlPageProvider` page provider needs to be replaced by  `SearchServicePageProvider`.

When an enhanced search client like `nuxeo-search-client-opensearch1` is installed it will be used.

For instance, this
```xml
<genericPageProvider class="org.nuxeo.elasticsearch.provider.ElasticSearchNxqlPageProvider"
```
Should be replaced by:
```xml
<genericPageProvider class="org.nuxeo.ecm.platform.query.nxql.SearchServicePageProvider"
```

## XML Contributions
Contributions made to `org.nuxeo.elasticsearch.ElasticSearchComponent` extension points should be updated to the new services. 

For instance `hint` contributions should be changed from
```xml
  <extension target="org.nuxeo.elasticsearch.ElasticSearchComponent" point="elasticSearchHints">
    <hint name="foo" class="..." />
```
to 
```xml
<extension target="org.nuxeo.ecm.core.search.client.opensearch1" point="hint">
  <hint name="foo" class="..." />
```

## Automation Operations

`ElasticSearch.Index` should be replaced by `Search.Index`.

`Elasticsearch.BulkIndex` should be replaced by `Search.Index`.

> `Elasticsearch.WaitForIndexing` should be replaced by `Search.WaitForIndexing`.

Reviewer: Maybe we should talk about operation parameters change?

Author: I have created https://hyland.atlassian.net/browse/NXP-33161 to accept the document model.

elasticsearch.indexNumberOfReplicas=1
elasticsearch.restClient.connectionTimeoutMs=30000
elasticsearch.restClient.socketTimeoutMs=121000
elasticsearch.restClient.ssl.certificate.verification=true

# optional
elasticsearch.restClient.username=
elasticsearch.restClient.password=
elasticsearch.restClient.truststore.path=
elasticsearch.restClient.truststore.password=
elasticsearch.restClient.truststore.type=
elasticsearch.restClient.keystore.path=
elasticsearch.restClient.keystore.password=
elasticsearch.restClient.keystore.type=
```

Now you will configure OpenSearch 1 / ElasticSearch 7 or 8 connection with:

```properties
# generic
nuxeo.opensearch1.client.server=http://localhost:9200

nuxeo.opensearch1.client.connectionTimeout=3s
nuxeo.opensearch1.client.socketTimeout=121s
nuxeo.opensearch1.client.sslCertificateVerification=true

# optional
nuxeo.opensearch1.client.username=
nuxeo.opensearch1.client.password=
nuxeo.opensearch1.client.trustStore.path=
nuxeo.opensearch1.client.trustStore.password=
nuxeo.opensearch1.client.trustStore.type=
nuxeo.opensearch1.client.keyStore.path=
nuxeo.opensearch1.client.keyStore.password=
nuxeo.opensearch1.client.keyStore.type=

> ```

Reviewer: What about adding translog durability, indexNumberOfShards and indexNumberOfReplicas as they are mentioned in the previous part about configuration parameter for LTS 2023?

Reviewer: Also, should we mention `nuxeo.search.client.default.opensearch1.index.name`?

---
title: How to upgrade to New Search Service
description: Instructions to upgrade your Nuxeo Project to LTS 2025 - Nuxeo Search Service Part
review:
   comment: ''
   date: '2025-04-21'
   status: ok
labels:
    - release-notes
toc: true
tree_item_index: 250
---

## Introduction

{{! excerpt}}
A new `SearchService` has been introduced in Nuxeo LTS 2025, the `ElasticSearchService` has been removed, representing a breaking change. The code needs to be modified to use the new `SearchService`.
{{! /excerpt}}

## Maven Dependency

`nuxeo-elasticsearch-core` has been removed in LTS 2025
```xml
<dependency>
  <groupId>org.nuxeo.elasticsearch</groupId>
  <artifactId>nuxeo-elasticsearch-core</artifactId>
</dependency>
```
Should be replaced by:
```xml
<dependency>
  <groupId>org.nuxeo.ecm.core</groupId>
  <artifactId>nuxeo-core-search</artifactId>
</dependency>
```

The management API provided by `nuxeo-elasticsearch-rest` is now part of `nuxeo-rest-api-server`.

> The management API supersedes the old `Elasticsearch.*` automation operation to reindex.

Reviewer 2: ```suggestion
The management API supersedes the old `Elasticsearch.*` automation operations to reindex.
```
In `Elasticsearch.*`, the `*` means many operations?

Author: yes I will rephrase.

## Java Code

The new `SearchService` implements a standardized method for executing NXQL queries against document repositories. Queries are executed on a dedicated search index, which can be contributed. This index primarily identifies repository and search client configurations.

By default, an index named `repository` is provided. This index is configured for the default repository and utilizes a Repository search client, leveraging the `CoreSession` to execute queries on the repository backend.

> When a Search Client package, such as `nuxeo-search-client-opensearch1` is installed to enable OpenSearch or Elasticsearch back-end functionality, an `enhanced` index is automatically created. This enhanced index will serve as the default search index for the default repository.

Reviewer 2: ```suggestion
When a Search Client package, such as `nuxeo-search-client-opensearch1`, is installed to enable OpenSearch or Elasticsearch back-end functionality, an `enhanced` index is automatically created. This enhanced index will serve as the default search index for the default repository.
```

elasticsearch.restClient.keystore.type=

# repository index
elasticsearch.indexName=nuxeo
elasticsearch.index.translog.durability=request
elasticsearch.indexNumberOfShards=5
elasticsearch.indexNumberOfReplicas=1
```

Now you will configure OpenSearch 1 / ElasticSearch 7 or 8 connection with:

```properties
# generic
nuxeo.opensearch1.client.server=http://localhost:9200
nuxeo.opensearch1.client.connectionTimeout=3s
nuxeo.opensearch1.client.socketTimeout=121s
nuxeo.opensearch1.client.sslCertificateVerification=true

# optional
nuxeo.opensearch1.client.username=
nuxeo.opensearch1.client.password=
nuxeo.opensearch1.client.trustStore.path=
nuxeo.opensearch1.client.trustStore.password=
nuxeo.opensearch1.client.trustStore.type=
nuxeo.opensearch1.client.keyStore.path=
nuxeo.opensearch1.client.keyStore.password=
nuxeo.opensearch1.client.keyStore.type=

# default index for the default repository
nuxeo.search.client.default.opensearch1.index.name=nuxeo
nuxeo.search.client.default.opensearch1.settings.indexTranslogDurability=request
nuxeo.search.client.default.opensearch1.settings.numberOfShards=5
nuxeo.search.client.default.opensearch1.settings.numberOfReplicas=1
```

> For now there is no managed alias support, the `elasticsearch.manageAlias.enabled` should not be used.

Reviewer 2: ```suggestion
From now on, there is no managed alias support, the `elasticsearch.manageAlias.enabled` should not be used.
```

Reviewer 2: Ok then `Currently, ` :)
At least, add a comma after `For now,`
